debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/RowDeserializers.java

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright Debezium Authors.
+ * 
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.Year;
+import java.time.ZoneOffset;
+import java.util.Calendar;
+import java.util.Map;
+
+import com.github.shyiko.mysql.binlog.event.TableMapEventData;
+import com.github.shyiko.mysql.binlog.event.deserialization.AbstractRowsEventDataDeserializer;
+import com.github.shyiko.mysql.binlog.event.deserialization.DeleteRowsEventDataDeserializer;
+import com.github.shyiko.mysql.binlog.event.deserialization.UpdateRowsEventDataDeserializer;
+import com.github.shyiko.mysql.binlog.event.deserialization.WriteRowsEventDataDeserializer;
+import com.github.shyiko.mysql.binlog.io.ByteArrayInputStream;
+
+/**
+ * Custom deserializers for the MySQL Binlog Client library.
+ * <p>
+ * A few of the {@link AbstractRowsEventDataDeserializer MySQL Binlog Client row deserializers} convert MySQL raw row data into
+ * {@link java.sql.Date}, {@link java.sql.Time}, and {@link java.sql.Timestamp} values using {@link Calendar} instances (and thus
+ * implicitly use the local timezone to calculate the milliseconds past epoch. Rather than perform this conversion, we prefer to
+ * convert the raw MySQL row values directly into {@link LocalDate}, {@link LocalTime}, {@link LocalDateTime}, and
+ * {@link OffsetDateTime}.
+ * <p>
+ * Unfortunately, all of the methods used to deserialize individual values are defined on the
+ * {@link AbstractRowsEventDataDeserializer} base class, and inherited by the {@link DeleteRowsEventDataDeserializer},
+ * {@link UpdateRowsEventDataDeserializer}, and {@link WriteRowsEventDataDeserializer} subclasses. Since we can't provide
+ * a new base class, the simplest way to override these methods is to subclass each of these 3 subclasses and override the
+ * methods on all 3 classes. It's ugly, but it works.
+ * <p>
+ * See the <a href="https://dev.mysql.com/doc/refman/5.0/en/datetime.html">MySQL Date Time</a> documentation.
+ * 
+ * @author Randall Hauch
+ */
+class RowDeserializers {
```

Comment:
This is an example of why subclasses sucks compared with composition, which would have made things far easier. There is non-trivial logic inside the `Delete*`, `Update*`, and `Write*` base classes, yet the code we want to override is in their base class `Abstract*`. So I could have either create a new base class and then copied over the logic to create three new concrete subclasses, or just subclassed the three `Delete*`, `Update*`, and `Write*` classes and overrode all of the methods in them. I chose to do the latter, but rather than replicate the logic in the overridden methods of those three classes, they now just delegate to static methods. My back is still hurting from the gymnastics I had to do to make it work.